client/rest-high-level/src/test/java/org/elasticsearch/client/BulkRequestWithGlobalParametersIT.java

```diff
@@ -204,12 +203,32 @@ public void testMixLocalAndGlobalRouting() throws IOException {
         Iterable<SearchHit> hits = searchAll(new SearchRequest("index").routing("globalRouting", "localRouting"));
         assertThat(hits, containsInAnyOrder(hasId("1"), hasId("2")));
     }
+    
+    public void testGlobalIndexNoTypes() throws IOException {
+        BulkRequest request = new BulkRequest("global_index");
+        request.add(new IndexRequest().id("1")
+            .source(XContentType.JSON, "field", "bulk1"));
+        request.add(new IndexRequest().id("2")
+            .source(XContentType.JSON, "field", "bulk2"));
 
-    private BulkResponse bulk(BulkRequest request) throws IOException {
-        BulkResponse bulkResponse = execute(request, highLevelClient()::bulk, highLevelClient()::bulkAsync);
+        bulk(request);
+
+        Iterable<SearchHit> hits = searchAll("global_index");
+        assertThat(hits, everyItem(hasIndex("global_index")));
+    }    
+
+    private BulkResponse bulkWithTypes(BulkRequest request) throws IOException {
+        BulkResponse bulkResponse = execute(request, highLevelClient()::bulk, highLevelClient()::bulkAsync, 
+                expectVersionSpecificWarnings((v)->{v.current(BulkRequest.TYPES_DEPRECATION_MESSAGE);}));
```

Comment:
One more instance of `expectVersionSpecificWarnings` that can be updated to `expectWarnings`.